### PR TITLE
is_chained + black

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by xiangyan99** [comment]
https://github.com/Azure/azure-sdk-for-java/pull/35771

**Review by pvaneck — COMMENTED** [review]
Looks like the keyword argument `is_chained` is now a part of the public API for these dev-time credentials. Was this is supposed to be an internal only param?

**Comment by xiangyan99** [comment]
> Looks like the keyword argument `is_chained` is now a part of the public API for these dev-time credentials. Was this is supposed to be an internal only param?

Because those credentials did not accept kwargs, to accept extra parameter, there is no way to hide it from public API changes.

But they are not breaking changes.

**Comment by pvaneck** [comment]
Is [`within_credential_chain`](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_internal/utils.py#L15) something we could use instead for this purpose?

Seems like we use it in quite a few places for determining the log-level depending on if the credential is within a chain or not ([example](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_internal/get_token_mixin.py#L89)).

**Comment by xiangyan99** [comment]
> Is [`within_credential_chain`](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_internal/utils.py#L15) something we could use instead for this purpose?
> 
> Seems like we use it in quite a few places for determining the log-level depending on if the credential is within a chain or not ([example](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_internal/get_token_mixin.py#L89)).

When I synced with Bill, we only wanted to swallow the error in DAC. But not in all chained credentials.

File: sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py [hunk 1]
```diff
@@ -90,21 +93,26 @@ def get_token(self, *scopes: str, **kwargs: Any) -> AccessToken:
...
         )
         if tenant:
             command += " --tenant " + tenant
-        output = _run_command(command, self._process_timeout)
+        output = _run_command(command, self._process_timeout, is_chained=self._is_chained)
 
         token = parse_token(output)
         if not token:
             sanitized_output = sanitize_output(output)
+            if self._is_chained:
```
Comment (pvaneck) [inline sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py:105]: I think it'd be worthwhile to dedupe the message strings in cases like this;

```python
message = "Unexpected output from Azure CLI: '{}'. \n" \
          "To mitigate this issue, please refer to the troubleshooting guidelines here at" \
          "https://aka.ms/azsdk/python/identity/azclicredential/troubleshoot.".format("santiized")
if self._is_chained:
    raise CredentialUnavailableError(message)
raise ClientAuthenticationError(message)
```

**Review by pvaneck — COMMENTED** [review]
Just a comment about deduping error messages.

I'm generally okay with the added `is_chained` keyword argument. If we do get pushback for introducing this param to the API, then I think there are ways around it like setting `credential._is_chained = True` directly after instantiating it in DAC.

**Review by pvaneck — APPROVED** [review]

File: sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py [hunk 2]
```diff
@@ -90,21 +93,21 @@ def get_token(self, *scopes: str, **kwargs: Any) -> AccessToken:
...
+        output = _run_command(command, self._process_timeout, is_chained=self._is_chained)
 
         token = parse_token(output)
         if not token:
             sanitized_output = sanitize_output(output)
-            raise ClientAuthenticationError(
-                message="Unexpected output from Azure CLI: '{}'. \n"
-                        "To mitigate this issue, please refer to the troubleshooting guidelines here at "
-                        "https://aka.ms/azsdk/python/identity/azclicredential/troubleshoot.".format(sanitized_output))
+            message = "Unexpected output from Azure CLI: '{}'. \n"
```
Comment (pvaneck) [inline sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py:105]: woops, looks like format is on the wrong line. Can maybe just f string the first line here?  `{sanitized_output}`

**Pull request merged**